lib/ui/index.js

```diff
@@ -194,11 +193,13 @@ UI.prototype.prompt = function(questions, callback) {
 
   // If no callback was provided, automatically return a promise
   if (callback) {
-    inquirer.prompt(questions, callback);
-  } else {
-    return new Promise(function(resolve) {
-      inquirer.prompt(questions, resolve);
+    inquirer.prompt(questions).then(function(result) {
+      callback(result);
+    }).catch(function() {
+      callback(null);
```

Comment:
I tend to agree, I'm just not sure *how* to rethrow in this case. The old `callback` argument is not using the node `function(err, result)` signature so we can't pass any errors to the callback like that.